man/man8/zfs.8

```diff
@@ -983,6 +1009,35 @@ This property can also be referred to by its shortened column name,
 
 .sp
 .ne 2
+.na
+\fB\fBencryption\fR=\fBoff\fR | \fBon\fR | \fBaes-128-ccm\fR | \fBaes-192-ccm\fR | \fBaes-256-ccm\fR | \fBaes-128-gcm\fR | \fBaes-192-gcm\fR | \fBaes-256-gcm\fR
+.ad
+.sp .6
+.RS 4n
+Controls the encryption cipher suite (block cipher, key length, and mode) used for this dataset. Requires the encryption feature to be enabled on the pool. Requires a keysource to be set at dataset creation time.
+.sp
+Selecting \fBencryption\fR=\fBon\fR when creating a dataset indicates that the current default encryption suite will be selected, which is currently \fBaes-256-ccm\fR. In order to provide consistent data protection, encryption must be specified at dataset creation time and it cannot be changed afterwards.
+.sp
+The \fBencryption\fR property is inherited by child datasets by default. All inheriting datasets will be accessible when the root key is mounted. A child dataset may also set encryption locally, which will indicate that it should have a separate key and \fBkeysource\fR from the parent. Regardless, all children of an encrypted dataset must also be encrypted.
+.sp
+For more details and caveats about encryption see \fBzpool-features\fR.
+.RE
+
+.sp
+.ne 2
+.mk
+.na
+\fB\fBkeysource\fR=<\fBraw\fR | \fBhex\fR | \fBpassphrase\fR>,<\fBprompt\fR | \fBfile://\fR\fI<absolute file path\fR>>
```

Comment:
The `pbkdf2iters` is actually a separate property, also managed with the `zfs key -c` command. I still like this separation especially as far as future proofing goes because (as I discussed in another comment) the keysource can be implemented such that it can be changed without needing the keys loaded. This means I could rewrite keysource setting to use `zfs set` instead.

As far as how compatible we should be with Solaris? That's a tough call. On one hand, its nice that for the most part our documentation in mostly portable and there have been a ton of times I've been looking up various zfs commands and found the Oracle documentation way before ours. On the other hand we are definitely not %100 compatible anyway for various security and asthetic reasons. I defer to @behlendorf